service/cluster/psmdb_cluster.go

```diff
@@ -112,7 +117,39 @@ func (s *PSMDBClusterService) CreatePSMDBCluster(ctx context.Context, req *contr
 
 // UpdatePSMDBCluster updates existing PSMDB cluster.
 func (s *PSMDBClusterService) UpdatePSMDBCluster(ctx context.Context, req *controllerv1beta1.UpdatePSMDBClusterRequest) (*controllerv1beta1.UpdatePSMDBClusterResponse, error) {
-	return nil, status.Error(codes.Unimplemented, s.p.Sprintf("This method is not implemented yet."))
+	client, err := k8sclient.New(ctx, req.KubeAuth.Kubeconfig)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	defer client.Cleanup() //nolint:errcheck
+
+	cluster, err := client.GetPSMDBCluster(ctx, req.Name)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot read cluster info")
+	}
+
+	// This is to prevent concurrent updates
+	if cluster.State != k8sclient.ClusterStateReady {
+		return nil, ErrPSMDBClusterNotReady //nolint:wrapcheck
+	}
+
+	params := &k8sclient.PSMDBParams{
+		Name: req.Name,
+		Size: req.Params.ClusterSize,
+	}
+	params.Replicaset = &k8sclient.Replicaset{
+		ComputeResources: &k8sclient.ComputeResources{
+			CPUM:        req.Params.Replicaset.ComputeResources.CpuM,
```

Comment:
Do we need to set CPUM only if its value bigger then 0?
and dame question for MemoryBytes.